src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/resources/script.js

```diff
@@ -45,50 +45,52 @@ function createElem(doc, tag, path) {
     scriptElement.parentNode.insertBefore(script, scriptElement);
 }
 
-function show(type) {
-    count = 0;
-    for (var key in data) {
-        var row = document.getElementById(key);
-        if ((data[key] &  type) !== 0) {
-            row.style.display = '';
-            row.className = (count++ % 2) ? rowColor : altColor;
-        }
-        else
-            row.style.display = 'none';
+function show(tableId, selected, columns) {
+    if (tableId !== selected) {
+        document.querySelectorAll('div.' + tableId + ':not(.' + selected + ')')
+            .forEach(function(elem) {
+                elem.style.display = 'none';
+            });
     }
-    updateTabs(type);
+    document.querySelectorAll('div.' + selected)
+        .forEach(function(elem, index) {
+            elem.style.display = '';
+            var isEvenRow = index % (columns * 2) < columns;
+            elem.classList.remove(isEvenRow ? rowColor : altColor);
+            elem.classList.add(isEvenRow ? altColor : rowColor);
```

Comment:
I assume that `elem.classList` is a set, not a list.  If it was previously an odd/even row and is still the same, we don't want to add an extra duplicate entry to the list.

In the DOM spec here
https://dom.spec.whatwg.org/#dom-element-classlist
I could only find non-normative narrative description that it is a set.

Reply:
Yes, Element.classList.add(...) has Set-semantics, it will not add the class again if it is already contained.